docs/standard/base-types/best-practices.md

```diff
@@ -69,7 +69,7 @@ manager: "wpickett"
   
 -   When developing a pattern, you should consider how backtracking might affect the performance of the regular expression engine, particularly if your regular expression is designed to process unconstrained input. For more information, see the [Take Charge of Backtracking](#Backtracking) section.  
   
--   Thoroughly test your regular expression using invalid and near-valid input as well as valid input. To generate input for a particular regular expression randomly, you can use [Rex](http://go.microsoft.com/fwlink/?LinkId=210756), which is a regular expression exploration tool from Microsoft Research.  
+-   Thoroughly test your regular expression using invalid and near-valid input as well as valid input. To generate input for a particular regular expression randomly, you can use [Rex](https://www.microsoft.com/research/project/rex-regular-expression-exploration/?from=http%3A%2F%2Fresearch.microsoft.com%2Fen-us%2Fprojects%2Frex%2F), which is a regular expression exploration tool from Microsoft Research.  
```

Comment:
a more cleaner url is https://www.microsoft.com/en-us/research/project/rex-regular-expression-exploration/